bundles/http/Citrix/0b4d1a6b-30f3-4134-a354-42fa008b757e/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,1029 @@
+let parser = library.load("xml2js")
+let jp = library.load("jsonpath-plus")
+const moment = library.load("moment-timezone")
+
+const createRequest = (client,session) => async (body) => {
+    try {
+        const response = await client.fetch('/wfc/XmlService', {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/xml",
+                Cookie: session
+            },
+            body: body
+        });
+        const data = await response.text();
+        const result = await parser.parseStringPromise(data);
+        let status = 200
+        if (response.ok && result?.Kronos_WFC.Response[0].$.Status == 'Failure') {
+            status = 400
+            throw new Error(result.Kronos_WFC.Response[0]);
+        }
+
+        return {
+            response,
+            result,
+            status
+           // or maybe even better: `result: result.Kronos_WFC.Response[0]` ?          
+        }
+    } catch (error) {
+        throw new Error(JSON.stringify(error))
+    }
+}
+
+async function logIn(client, userName, password) {
+    let session = [];
+    const response = await client.fetch('/wfc/XmlService', {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/xml",
+            "Accept": "application/xml"
+        },
+        body: `<KRONOS_WFC VERSION="1.0"><REQUEST OBJECT="SYSTEM" ACTION="LOGON" USERNAME="${userName}" PASSWORD="${password}"/></KRONOS_WFC>`
+    });
+
+    const cookies = await response.headers.get('set-cookie');
+    const data = await response.text();
+    parser.parseString(data, (err , result)=>{
+        if(err != null){
+            throw new Error(JSON.stringify(err))
+        }
+        else if(result.Kronos_WFC.Response[0].$.Status == 'Failure'){
+            throw new Error(result.Kronos_WFC.Response[0].$.Message)
+        }
+
+        session = cookies.split(',').filter(value =>{
+            return value.indexOf('JSESSIONID') === 0
+        })
+    })
+    return session[0];
+}
+
+async function syncPeople(dataStore, client, session, integrationParameters) {
+    const month = moment().format('M')
+    const start = moment().month(month-1).startOf('month').format(integrationParameters.dateFormat)
+    const end = moment().month(month).endOf('month').format(integrationParameters.dateFormat)
+    const body =  `<?xml version="1.0"?>
+    <KRONOS_WFC VERSION="1.0"><Request Action = "RunQuery">
+    <HyperfindQuery HyperfindQueryName = "All Home"  QueryDateSpan = "${start} - ${end}" VisibilityCode= "Public"  QueryPersonOrEmployee="Person" />
```

Comment:
1. **HyperfindQuery:** As per Kronos API Doc  this is the predefined xml request  to get the All people Information across the Kronos SoR .  as a default query it requires query **QueryDateSpan ** as a parameter to fetch all people data.
2.Kronos  API   Request with   **QueryDateSpan**  will give same results if we pass the start and end time parameters. It will not make any difference in Incremental Sync and full Sync while **limiting** the time period.

Please refer Page: 1046 in below document for reference.
**API DOC:** https://citrix.sharepoint.com/sites/Product/APAC/DeveloperEcosystem/Shared%20Documents/Forms/AllItems.aspx?originalPath=aHR0cHM6Ly9jaXRyaXguc2hhcmVwb2ludC5jb20vOmY6L3MvUHJvZHVjdC9BUEFDL0RldmVsb3BlckVjb3N5c3RlbS9FdmZRbllaX0ZWaEF0QWo1TFhlNjZKRUItT2EzenFCNkw1azVjN3ZwRU9NblFnP3J0aW1lPV9TM001VlR6MkVn&viewid=21381dc4%2D0fb7%2D4aba%2D8195%2Daf2f1f9d5361&id=%2Fsites%2FProduct%2FAPAC%2FDeveloperEcosystem%2FShared%20Documents%2FLearning%2DTraining%2FProduct%2FAtmecs%2FKronos%2Fkronos%5Fworkforce%5Fcentral%2FKronos%20Docs%2FWFC813%5FTimekeeperDevelopersToolkitRef%2Epdf&parent=%2Fsites%2FProduct%2FAPAC%2FDeveloperEcosystem%2FShared%20Documents%2FLearning%2DTraining%2FProduct%2FAtmecs%2FKronos%2Fkronos%5Fworkforce%5Fcentral%2FKronos%20Docs